presto-main/src/main/java/io/prestosql/sql/InterpretedFunctionInvoker.java

```diff
@@ -92,18 +104,26 @@ public Object invoke(ResolvedFunction function, ConnectorSession session, List<O
         }
     }
 
-    private static MethodHandle bindInstanceFactory(MethodHandle method, ScalarFunctionImplementation implementation)
+    private static InvocationConvention getInvocationConvention(ResolvedFunction function, FunctionMetadata functionMetadata)
```

Comment:
This method is not trivial, so consider factoring out this implementation and the one in ExpressionOptimizer into one that can be called from both places.